controllers/keystoneapi_controller.go

```diff
@@ -195,8 +196,22 @@ func (r *KeystoneAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *KeystoneAPIReconciler) reconcileDelete(ctx context.Context, instance *keystonev1.KeystoneAPI, helper *helper.Helper) (ctrl.Result, error) {
 	r.Log.Info("Reconciling Service delete")
 
+	// remove db finalizer before the keystone one
+	//get db pointer, then in database use ctlr.ult to remove it's finalizer
+
+	//TODO1 figure out where to get the service Name and Namepace for r.
+	// maybe use the func - GetKeystoneServiceWithName
+
+	serviceDb, err := database.GetServiceDbByName(ctx, helper, instance.Name, instance.Namespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	database.Delete(serviceDb, ctx, helper)
```

Comment:
I think we do not need a func for this. could just do it like for the for keystoneservice/endpoint https://github.com/openstack-k8s-operators/placement-operator/blob/master/controllers/placementapi_controller.go#L210-L222

Reply:
ok done